server/db/driver/pg/internal/epochs.go

```diff
@@ -61,4 +61,39 @@ const (
 		FROM %s
 		WHERE epoch_end >= $1
 		ORDER BY epoch_end;`
+
+	// CreateEpochReportTable creates an candles table that holds binned
+	// candle data.
+	CreateCandlesTable = `CREATE TABLE IF NOT EXISTS %s (
+		end_stamp INT8,
+		candle_dur INT8,
+		match_volume INT8,
+		quote_volume INT8,
+		high_rate INT8,
+		low_rate INT8,
+		start_rate INT8,
+		end_rate INT8
+	);`
+
+	CreateCandlesIndex = `CREATE INDEX IF NOT EXISTS %s ON %s (end_stamp, candle_dur);`
+
+	InsertCandle = `INSERT INTO %s (end_stamp, candle_dur, match_volume,
+		quote_volume, high_rate, low_rate, start_rate, end_rate
+	)
+	VALUES ($1, $2, $3, $4, $5, $6, $7, $8);`
+	// How to ON CONFLICT (end_stamp, candle_dur) DO NOTHING ?
```

Comment:
Not really sure yet how to get this to work. If I try this `ON CONFLICT` clause, I get a db error
```
[ERR] MKT: Error updating API data collector: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification
```
Or maybe the two-column index isn't worth it. The PostgreSQL docs say to use multi-column indexes sparingly.

Reply:
Why not just have three tables, one for each candle duration?